.github/workflows/run-standard-tests.yml

```diff
@@ -126,59 +162,61 @@ jobs:
         run: |
+          mkdir results
           if [[ "${{ matrix.config }}" == "release" ]] ; then
-            python3 $APPLY_PY errors test/LearningTest/TestKhiops/Standard | tee test/LearningTest/TestKhiops/Standard/errors.txt
+            python $APPLY_PY test/LearningTest/TestKhiops/Standard errors | tee -a results/errors.txt
             if [[ "${{ matrix.running-mode }}" != "parallel" ]] ; then
-              python $APPLY_PY errors test/LearningTest/TestCoclustering/Standard | tee test/LearningTest/TestCoclustering/Standard/errors.txt
-              python $APPLY_PY errors test/LearningTest/TestKNITransfer/Standard | tee test/LearningTest/TestKNITransfer/Standard/errors.txt
-              python $APPLY_PY errors test/LearningTest/TestKNITransfer/MultiTables SpliceJunction | tee test/LearningTest/TestKNITransfer/MultiTables/errors.txt
+              python $APPLY_PY test/LearningTest/TestCoclustering/Standard errors | tee -a results/errors.txt
+              python $APPLY_PY test/LearningTest/TestKNI/Standard errors | tee -a results/errors.txt
             fi
           else
-            python3 $APPLY_PY errors test/LearningTest/TestKhiops/Standard IrisLight | tee test/LearningTest/TestKhiops/Standard/errors.txt
+            python $APPLY_PY test/LearningTest/TestKhiops/Standard/IrisLight errors | tee -a results/errors.txt
             if [[ "${{ matrix.running-mode }}" != "parallel" ]] ; then
-              python $APPLY_PY errors test/LearningTest/TestCoclustering/Standard Iris | tee test/LearningTest/TestCoclustering/Standard/errors.txt
-              python $APPLY_PY errors test/LearningTest/TestKNITransfer/Standard Iris | tee test/LearningTest/TestKNITransfer/Standard/errors.txt
+              python $APPLY_PY test/LearningTest/TestCoclustering/Standard/Iris errors | tee -a results/errors.txt
+              python $APPLY_PY test/LearningTest/TestKNI/Standard/Iris errors | tee -a results/errors.txt
             fi
           fi
+          python $COLLECT_PY test/LearningTest/ results --collect-type warnings -f basic
       - name: Check results
+        # We escape eol in the results file to print multilines in the github UI https://github.com/orgs/community/discussions/26288
         shell: bash
         run: |
-          if grep -qr "error" --include="errors.txt" test/LearningTest/ ; then
-            echo "::error::Errors in test"
+          MSG=$(<results/errors.txt)
+          MSG="${MSG//'%'/'%25'}"
+          MSG="${MSG//$'\n'/'%0A'}"
+          MSG="${MSG//$'\r'/'%0D'}"
+          if grep -q -e "error" -e "The test has not been launched" results/errors.txt ; then
+            echo "::error::$MSG"
             false
           fi
-          if grep -qr "The test has not been launched" --include="errors.txt" test/LearningTest/ ; then
-            echo "::error::Test not launched"
+          if [ "${{ github.event.inputs.warning_as_error }}" == "true" ] && grep -q "warning" results/errors.txt  ; then
```

Comment:
You may use `inputs.warnings_as_error` instead of `github.event.inputs.warnings_as_error`